### PR TITLE
Mainline aquaero

### PR DESCRIPTION
**Pull request by leoratte**
I took the liberty to create a patchset for aquaero sensor support. I have not sent this upstream yet because I want to get your opinion first.

### CODE REVIEW

**Comment by aleksamagicka** [comment]
Thanks! I'll review ASAP.

File: docs/aquacomputer_d5next.rst [hunk 1]
```diff
@@ -19,6 +20,9 @@ Description
 This driver exposes hardware sensors of listed Aquacomputer devices, which
 communicate through proprietary USB HID protocols.
 
+The Aquaero devices expose eight temperature sensors and four PWM controllable
```
Comment (aleksamagicka) [inline docs/aquacomputer_d5next.rst:23]: Also mention virtual temp sensors.

**Review by aleksamagicka — COMMENTED** [review]
Third commit: `This allows reading the fan sensors for devices with different offsets with the same code.` That's a bit ambiguous, write something like this, so it's clear what benefits from it: `This allows reading fan sensors  in aqc_raw_event() from devices which use a different sensor layout in their status HID report.`.

----
Generally looks good, I'll look over all again tomorrow hopefully and test just to be sure.

**Comment by leoratte** [comment]
Thanks for your feedback.
Could you bring the hwmon-state branch up to the actual upstream state, especially the docs are wildly out of date.

**Comment by aleksamagicka** [comment]
True, just updated it.

**Comment by aleksamagicka** [comment]
Why is https://github.com/aleksamagicka/aquacomputer_d5next-hwmon/pull/61/commits/5d78e45ed69b75abf5b9ad333618cc265ffc2f71 needed?

Other commits LGTM. Do you have a cover letter in mind?

**Comment by leoratte** [comment]
5d78e45 fixes the last difference between mainline and this branch. Also editors like `vim` add this by default so it is standard in all kernel files.
Checkpatch says everything is fine and for the cover letter i would use:
```
Add support for the Aquacomputer Aquaero 5/6 fan controllers. These fan
controllers support 4 fans, 8 physical temperature sensors, 8 virtual
temperature sensors and 2 flow sensors.

The first 5 patches prepare supporting Aquacomputer Aquaero without
changing the functionality. Patch 6 adds support for Aquaero sensors.
Fan control is planned for a future patch.
```

**Comment by aleksamagicka** [comment]
Ah, never looked at the file endings. 👍 

In the second paragraph, I'd say `prepare support for`, it sounds a bit more natural I think.

Other than that, I think this is good for a first revision. Should Guenter have complaints, we'll resolve them.

**Comment by aleksamagicka** [comment]
This just made it in, merging.

**Comment by leoratte** [comment]
Nice

**Pull request merged**